parser/generator.go

```diff
@@ -162,6 +162,7 @@ func (g *Generator) GenerateDockerfile() error {
 
 	for _, phase := range phases {
 		if len(phase.commands) > 0 {
+			bufferRun.WriteString("source /.bzkenv\n")
```

Comment:
Yep, I was thinking in the same line: add a boolean variable and set it to true when a phase has some commands, and only source the env file if the variable is set.